rest-service/manager_rest/rest/resources_v3_1/plugins.py

```diff
@@ -70,3 +75,85 @@ def post(self, **kwargs):
         )
         with rest_utils.skip_nested_marshalling():
             return super(Plugins, self).post(visibility=visibility)
+
+
+class PluginsUpdate(SecuredResource):
+
+    @rest_decorators.exceptions_handled
+    @authorize('plugins_update_create')
+    @rest_decorators.marshal_with(models.PluginsUpdate)
+    def post(self, id, phase):
+        """
+        Supports two stages of a plugin update.
+        Phases:
+            1. Creates a temporary blueprint and executes a deployment update
+            (will update only the plugins) for all the deployments of the
+            given blueprint.
+            2. Updates the original blueprint plan and deletes the temp one.
+
+        :param id: the blueprint ID to update it's deployments' plugins if
+        phase == 'initiate', otherwise (phase == 'finalize') the plugin update
```

Comment:
initiate is different from INITIAL, is it a mistake? I think it's better to not use the const values because they can change and the docstring could be not changed with

Reply:
The phase is the initial phase but you ask the server to initiate

Reply:
I changed the docstring to contain the constant name